lua/rustaceanvim/config/internal.lua

```diff
@@ -275,6 +276,10 @@ local RustaceanDefaultConfig = {
     cmd = function()
       return { 'rust-analyzer', '--log-file', RustaceanConfig.server.logfile }
     end,
+
+    ---@type fun(filename: string):string|nil
+    root_dir = cargo.get_root_dir,
```

Comment:
While I agree that it's generally better to make this a function in user configs, I still see a use case for allowing it to be a string.

- It improves Vimscript interoperability (one of the reasons I prefer `vim.g` for configuration).
- There's no need for it to be a function when setting it in `:help exrc`.